### PR TITLE
feat(amplify-codegen-appsync-model-plugin): update java model generator

### PR DESCRIPTION
**Pull request by yuth**
Updated model generator to
1. Generate a class loader
2. Expose an builder step for id
3. Update constructor of Model class to accpet individual field arguments
4. Updated the build method

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 1]
```diff
@@ -176,10 +287,13 @@ export class AppSyncModelJavaVisitor<
 
     // methods
     // build();
+    const buildImplementation = [`String id = this.id != null ? this.id : UUID.randomUUID().toString();`, ''];
+    const buildParams = model.fields.map(field => this.getFieldName(field)).join(',\n');
+    buildImplementation.push(`return new ${this.getModelName(model)}(\n${indentMultiline(buildParams)});`);
```
Comment (nikhname) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]: Can this be a constant?

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 2]
```diff
@@ -138,8 +244,13 @@ export class AppSyncModelJavaVisitor<
...
+    // build method
     builderBody.push(`${this.getModelName(model)} build();`);
-    nullableFields.forEach((field, idx) => {
+
+    // id method. Special case as this can throw exception
+    builderBody.push(`${this.getStepInterfaceName('Build')} id(String id) throws AmplifyException;`);
+
+    nullableFields.forEach(field => {
       const fieldName = this.getFieldName(field);
       builderBody.push(`${this.getStepInterfaceName('Build')} ${fieldName}(${this.getNativeType(field)} ${fieldName});`);
```
Comment (nikhname) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]: Could this argument be a constant?

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 3]
```diff
@@ -138,8 +244,13 @@ export class AppSyncModelJavaVisitor<
       .withName(this.getStepInterfaceName('Build'))
       .access('public');
     const builderBody = [];
+    // build method
     builderBody.push(`${this.getModelName(model)} build();`);
-    nullableFields.forEach((field, idx) => {
+
+    // id method. Special case as this can throw exception
+    builderBody.push(`${this.getStepInterfaceName('Build')} id(String id) throws AmplifyException;`);
```
Comment (nikhname) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]: Could this argument be a constant?

File: packages/amplify-codegen-appsync-model-plugin/src/languages/java-declaration-block.ts [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+      }
+
+      result += `${annotatesStr}${this._access}${isStatic}${final} ${this._kind} ${name}${extendStr}${implementsStr} `;
+    }
+
+    const members = this._members.length
+      ? indentMultiline(stripIndent(this._members.map(member => this.printMember(member) + ';').join('\n')))
+      : null;
+    const methods = this._methods.length
+      ? indentMultiline(stripIndent(this._methods.map(method => this.printMethod(method)).join('\n\n')))
```
Comment (nikhname) [inline packages/amplify-codegen-appsync-model-plugin/src/languages/java-declaration-block.ts]: Are the two new lines intentional?

File: packages/amplify-codegen-appsync-model-plugin/src/preset.ts [hunk 5]
```diff
@@ -2,9 +2,10 @@ import { Types } from '@graphql-codegen/plugin-helpers';
 import { Kind, TypeDefinitionNode } from 'graphql';
 import { join } from 'path';
 import { JAVA_SCALAR_MAP, SWIFT_SCALAR_MAP, TYPESCRIPT_SCALAR_MAP } from './scalars';
+import { LOADER_CLASS_NAME } from './configs/java-config';
 
 const APPSYNC_DATA_STORE_CODEGEN_TARGETS = ['java', 'android', 'swift', 'ios', 'javascript', 'typescript'];
-const JAVA_PACKAGE_NAME = 'com.amplify.datastore.generated';
+const JAVA_PACKAGE_NAME = 'com.amplifyframework.datastore.generated.model';
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/preset.ts]: can we define it once, then reference? reused string literals should be shared (applies to others as well)

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 6]
```diff
@@ -225,6 +339,42 @@ export class AppSyncModelJavaVisitor<
...
+    const idBuildStepBody = dedent`this.id = id;
+
+    try {
+        UUID.fromString(id); // Check that ID is in the UUID format - if not an exception is thrown
+    } catch (Exception exception) {
+        throw new AmplifyException("Model IDs must be unique in the format of UUID.",
+                exception,
+                "If you are creating a new object, leave ID blank and one will be autogenerated for you. " +
+                "Otherwise, if you are referencing an existing object, be sure you are getting the correct " +
+                "id for it. It's also possible you are referring to an item created outside of Amplify. We " +
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]: "It is currently not supported"

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 7]
```diff
@@ -225,6 +339,42 @@ export class AppSyncModelJavaVisitor<
...
+                "currently do not support this.",
+                false);
+    }
+
+    return this;`;
+
+    const idComment = dedent`WARNING: Do not set ID when creating a new object. Leave this blank and one will be auto generated for you.
```
Comment (attilah) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]: "auto generated" vs. "autogenerated". Above "autogenerated" is used, please make sure we're consistent.

File: packages/amplify-codegen-appsync-model-plugin/src/languages/java-declaration-block.ts (hunk 4 above)
Comment (yuth) [reply packages/amplify-codegen-appsync-model-plugin/src/languages/java-declaration-block.ts]: one is `members` and  the other one is `methods`

**Pull request merged**